File: Programmers/C++/Level_1/Lv1_체육복.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    vector<int> students(n, 1);
    
    for (int i=0; i<reserve.size(); i++)
        students[reserve[i]-1]++;
    for (int i=0; i<lost.size(); i++)
        students[lost[i]-1]--;
    
    for (int i=0; i<n; i++) {
        if (i > 0 && students[i] == 0 && students[i-1] == 2) {
            students[i-1]--;
            students[i]++;
        }
        if (i < n-1 && students[i] == 0 && students[i+1] == 2) {
            students[i+1]--;
            students[i]++;
        }
    }
    for (int s : students) {
        if (s > 0)
            answer++;
    }
    return answer;
}

File: Programmers/Python/Level_1/Lv1_체육복.py
def solution(n, lost, reserve):
    reserve_set = set(reserve) - set(lost)
    lost_set = set(lost) - set(reserve)
    
    for i in reserve_set:
        if i-1 in lost_set:
            lost_set.remove(i-1)
        elif i+1 in lost_set:
            lost_set.remove(i+1)
    
    return n-len(lost_set)